### ISSUE TITLE
NPE in KILtoSMTLib.renameSort without --smt none

### ISSUE THREAD

**Issue by bmmoore** [kframework/k-legacy#1692]
Trying to run `sum.imp` under the kore IMP++ from `lesson_8` on the tutorial-kore branch (at  4b1f3935c84db5941451fa443f86c9a83183254c) as
`krun --debug --kore --parser "kast --kore --sort Stmts" ../../2_imp/lesson_1/tests/sum.imp` throws

```
java.lang.NullPointerException
        at org.kframework.backend.java.symbolic.KILtoSMTLib.renameSort(KILtoSMTLib.java:368)
        at org.kframework.backend.java.symbolic.KILtoSMTLib.getSortAndFunctionDeclarations(KILtoSMTLib.java:246)
        at org.kframework.backend.java.symbolic.KILtoSMTLib.translateConstraint(KILtoSMTLib.java:176)
        at org.kframework.backend.java.symbolic.SMTOperations.checkUnsat(SMTOperations.java:40)
        at org.kframework.backend.java.symbolic.ConjunctiveFormula.checkUnsat(ConjunctiveFormula.java:580)
        at org.kframework.backend.java.kil.ConstrainedTerm.unify(ConstrainedTerm.java:224)
        at org.kframework.backend.java.symbolic.SymbolicRewriter.computeRewriteStepByRule(SymbolicRewriter.java:206)
        at org.kframework.backend.java.symbolic.SymbolicRewriter.lambda$computeRewriteStep$86(SymbolicRewriter.java:154)
        at org.kframework.backend.java.symbolic.SymbolicRewriter$$Lambda$81/976725249.apply(Unknown Source)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
        at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:529)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:516)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
        at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:469)
        at org.kframework.backend.java.symbolic.SymbolicRewriter.computeRewriteStep(SymbolicRewriter.java:163)
        at org.kframework.backend.java.symbolic.SymbolicRewriter.rewrite(SymbolicRewriter.java:80)
        at org.kframework.backend.java.symbolic.InitializeRewriter$SymbolicRewriterGlue.execute(InitializeRewriter.java:117)
        at org.kframework.krun.modes.KRunExecutionMode.execute(KRunExecutionMode.java:46)
        at org.kframework.krun.KRun.run(KRun.java:73)
        at org.kframework.krun.KRunFrontEnd$NewKRunFrontEnd.run(KRunFrontEnd.java:111)
        at org.kframework.krun.KRunFrontEnd.run(KRunFrontEnd.java:127)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:52)
        at org.kframework.main.Main.runApplication(Main.java:109)
        at org.kframework.main.Main.runApplication(Main.java:99)
        at org.kframework.main.Main.main(Main.java:51)
```

**Comment by andreistefanescu** [comment]
@bmmoore can you check if this is still an issue after #1712, please?

**Comment by bmmoore** [comment]
Seems to be fixed.

**Comment by daejunpark** [comment]
Recurred. (5f2ba5e227aca7be473395e30deca3a940dcd083)

```
$ pwd
/Users/daejunpark/kore/k/k-distribution/tutorial/1_k/3_lambda++/lesson_3
$ krun  --kore --parser 'kast --kore --sort Exp' ../../1_lambda/lesson_5/tests/arithmetic-div-zero.lambda --depth 5
java.lang.NullPointerException
    at org.kframework.backend.java.symbolic.KILtoSMTLib.renameSort(KILtoSMTLib.java:368)
    at org.kframework.backend.java.symbolic.KILtoSMTLib.getSortAndFunctionDeclarations(KILtoSMTLib.java:246)
    at org.kframework.backend.java.symbolic.KILtoSMTLib.translateConstraint(KILtoSMTLib.java:176)
    at org.kframework.backend.java.symbolic.SMTOperations.checkUnsat(SMTOperations.java:40)
    at org.kframework.backend.java.symbolic.ConjunctiveFormula.checkUnsat(ConjunctiveFormula.java:580)
    at org.kframework.backend.java.kil.ConstrainedTerm.unify(ConstrainedTerm.java:224)
    at org.kframework.backend.java.symbolic.SymbolicRewriter.computeRewriteStepByRule(SymbolicRewriter.java:206)
    at org.kframework.backend.java.symbolic.SymbolicRewriter.lambda$computeRewriteStep$81(SymbolicRewriter.java:154)
    at org.kframework.backend.java.symbolic.SymbolicRewriter$$Lambda$79/1827710791.apply(Unknown Source)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
    at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
    at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:529)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:516)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
    at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:469)
    at org.kframework.backend.java.symbolic.SymbolicRewriter.computeRewriteStep(SymbolicRewriter.java:163)
    at org.kframework.backend.java.symbolic.SymbolicRewriter.rewrite(SymbolicRewriter.java:80)
    at org.kframework.backend.java.symbolic.InitializeRewriter$SymbolicRewriterGlue.execute(InitializeRewriter.java:117)
    at org.kframework.krun.modes.KRunExecutionMode.execute(KRunExecutionMode.java:45)
    at org.kframework.krun.KRun.run(KRun.java:72)
    at org.kframework.krun.KRunFrontEnd$NewKRunFrontEnd.run(KRunFrontEnd.java:111)
    at org.kframework.krun.KRunFrontEnd.run(KRunFrontEnd.java:127)
    at org.kframework.main.FrontEnd.main(FrontEnd.java:52)
    at org.kframework.main.Main.runApplication(Main.java:109)
    at org.kframework.main.Main.runApplication(Main.java:99)
    at org.kframework.main.Main.main(Main.java:51)
```

**Comment by daejunpark** [comment]
Fixed via #1743
